invenio_rdm_records/resources/serializers/csl/schema.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2021 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""CSL based Schema for Invenio RDM Records."""
+
+from invenio_access.permissions import system_identity
+from invenio_vocabularies.proxies import current_service as vocabulary_service
+from marshmallow import Schema, fields, missing
+from marshmallow_utils.fields import SanitizedUnicode
+
+
+class CSLCreatorSchema(Schema):
+    """Creator/contributor common schema."""
+
+    family = fields.Str(attribute="person_or_org.name")
+
+
+class CSLJSONSchema(Schema):
+    """CSL Marshmallow Schema."""
+
+    id_ = SanitizedUnicode(data_key="id", attribute='id')
+    type_ = fields.Method('get_type', data_key="type")
+    title = SanitizedUnicode(attribute="metadata.title")
+    abstract = SanitizedUnicode(attribute="metadata.description")
+    author = fields.List(
+        fields.Nested(CSLCreatorSchema), attribute='metadata.creators')
+    issued = fields.Method('get_issued')
+    language = fields.Method('get_language')
+    version = SanitizedUnicode(attribute="metadata.version")
+    note = fields.Method('get_note')
+    doi = fields.Method('get_doi', data_key="DOI")
+    isbn = fields.Method('get_isbn', data_key="ISBN")
+    issn = fields.Method('get_issn', data_key="ISSN")
+    publisher = SanitizedUnicode(attribute='metadata.publisher')
+
+    def _read_resource_type(self, id_):
+        """Retrieve resource type record using service."""
+        return vocabulary_service.read(
+            ('resource_types', id_),
+            system_identity
+        )._record
+
+    def get_type(self, obj):
+        """Get resource type."""
+        resource_type = obj["metadata"]["resource_type"]
+        resource_type_record = self._read_resource_type(resource_type["id"])
+        props = resource_type_record["props"]
+        return props.get("csl", "article")  # article is CSL "Other"
+
+    def get_issued(self, obj):
+        """Get issued dates."""
+        date_parts = []
+        publication_date = obj["metadata"]["publication_date"].split("/")
```

Comment:
Question: the publication date format might not only with `-`. The allowed format is [EDTF level 0](https://www.loc.gov/standards/datetime/).
In the deposit form, it is written: `In case your upload was already published elsewhere, please use the date of the first publication. Format: YYYY-MM-DD, YYYY-MM, or YYYY. For intervals use DATE/DATE, e.g. 1939/1945.`

I would suggest to parse it then? See the parsing [here](https://github.com/inveniosoftware/invenio-rdm-records/blob/e47854aead0783175c7fc99ef05850e22178d4f0/invenio_rdm_records/resources/serializers/datacite/schema.py#L221):
```
parsed_date = parse_edtf(publication_date)
```

Reply:
https://github.com/inveniosoftware/invenio-rdm-records/issues/651